controllers/payment.controllers.js

```diff
@@ -0,0 +1,73 @@
+const Payment=require("../models/payment.models")
+const Booking=require("../models/booking.model")
+const constants=require("../utils/constants")
+
+exports.initPayment=async (req,res)=>{
+    try
+    {
+        const data={
+            bookedId:req.body.bookedId,
+            amount:req.body.amount,
+            status:req.body.status
+        }    
+       
+        const paymentObj=await Payment.create(data);
+       
+        const booking=await Booking.findOne({_id:paymentObj.bookedId})
+       
+        res.status(201).send(paymentObj)
+
+        return setTimeout(async()=>{
```

Comment:
No need of this code... Check for the booking status can be done in the MW itself